createFolders.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+PROJECT_NAME=data-and-algorithms
+DIRECTORY=${PWD}
+BASE_DIRECTORY=$(echo "$DIRECTORY" | rev | cut -d "/" -f1 | rev)
+
+if [[("$1" != "")]]; then
+   if [[("$BASE_DIRECTORY" != "$PROJECT_NAME")]]; then
+      echo "Make sure you are in $PROJECT_NAME project"
+      exit
+   fi
+   rootfileName=$1
+   mkdir ./$rootfileName
```

Comment:
here you can do

```shell
mkdir "${DIRECTORY}/${rootfileName}"
```

you should always try to brace the variables to make it super clear where they end. otherwise things like `$myvar_`would confuse readers. is the intention to be a variable called `myvar_` or is it `$myvar` followed by an underscore?

also, by using the `rsync` or `find` alternatives sugegsted below, this `mkdir` is unneeded.